integreat_cms/cms/utils/external_calendar_utils.py

```diff
@@ -0,0 +1,263 @@
+"""
+Utility functions for working with external calendars
+"""
+
+from __future__ import annotations
+
+import dataclasses
+import datetime
+import logging
+
+import icalendar.cal
+from django.utils.translation import gettext as _
+
+from integreat_cms.cms.constants import status
+from integreat_cms.cms.forms import EventForm, EventTranslationForm
+from integreat_cms.cms.models import EventTranslation, ExternalCalendar
+from integreat_cms.cms.utils.content_utils import clean_content
+
+
+# pylint: disable=too-many-instance-attributes
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class IcalEventData:
+    """
+    Stores data extracted from ical events
```

Comment:
This might cause confusion over whether we long time *store* this. We don't at least not in this form, only when it was put into an Event.
```suggestion
    Holds data extracted from ical events
```